urllib3/util/selectors.py

```diff
@@ -333,206 +333,227 @@ def select(self, timeout=None):
+                event_mask = 0
+                if events & EVENT_READ:
+                    event_mask |= select.POLLIN
+                if events & EVENT_WRITE:
+                    event_mask |= select.POLLOUT
+                self._poll.register(key.fd, event_mask)
+                return key
+
+            def unregister(self, fileobj):
+                key = super(PollSelector, self).unregister(fileobj)
+                self._poll.unregister(key.fd)
+                return key
+
+            def _wrap_poll(self, timeout=None):
+                """ Wrapper function for select.poll.poll() so that
+                _syscall_wrapper can work with only seconds. """
+                if timeout is not None:
+                    if timeout <= 0:
+                        timeout = 0
+                    else:
+                        # select.poll.poll() has a resolution of 1 millisecond,
+                        # round away from zero to wait *at least* timeout seconds.
+                        timeout = math.ceil(timeout * 1e3)
 
-        def _wrap_poll(self, timeout=None):
-            """ Wrapper function for select.poll.poll() so that
-            _syscall_wrapper can work with only seconds. """
-            if timeout is not None:
-                if timeout <= 0:
-                    timeout = 0
-                else:
-                    # select.poll.poll() has a resolution of 1 millisecond,
-                    # round away from zero to wait *at least* timeout seconds.
-                    timeout = math.ceil(timeout * 1e3)
+                result = self._poll.poll(timeout)
+                return result
 
-            result = self._poll.poll(timeout)
-            return result
+            def select(self, timeout=None):
+                ready = []
+                fd_events = _syscall_wrapper(self._wrap_poll, True, timeout=timeout)
+                for fd, event_mask in fd_events:
+                    events = 0
+                    if event_mask & ~select.POLLIN:
+                        events |= EVENT_WRITE
+                    if event_mask & ~select.POLLOUT:
+                        events |= EVENT_READ
 
-        def select(self, timeout=None):
-            ready = []
-            fd_events = _syscall_wrapper(self._wrap_poll, True, timeout=timeout)
-            for fd, event_mask in fd_events:
-                events = 0
-                if event_mask & ~select.POLLIN:
-                    events |= EVENT_WRITE
-                if event_mask & ~select.POLLOUT:
-                    events |= EVENT_READ
+                    key = self._key_from_fd(fd)
+                    if key:
+                        ready.append((key, events & key.events))
 
-                key = self._key_from_fd(fd)
-                if key:
-                    ready.append((key, events & key.events))
+                return ready
 
-            return ready
+    except OSError:
+        pass
```

Comment:
These except blocks are way too big. It'll be better to use `try...except...else` and place the class declaration in the `else` statement.